### PR TITLE
Prevent userSelected action to be called when the drop down is not shown

### PR DESCRIPTION
**Pull request by thangdinh**
Problem: `userDidSelect` is called even when the dropdown is not shown. The reason is we use the same result item inside the toggle button and clicking on it is equivalent to clicking on one item in the dropdown, which then triggers the `userDidSelect`.
Solution: Override the `click` function inside the `selectedItemView` which is the item displayed inside the toggle button.

Paired with @philpee2 . Please help me review this PR, @embooglement @Addepar/fire

### CODE REVIEW

File: src/select.coffee [hunk 1]
```diff
@@ -209,6 +209,13 @@ Ember.AddeparMixins.ResizeHandlerMixin, Ember.Widgets.KeyboardHelper,
       tagName: 'span'
       labelPath: Ember.computed.alias 'controller.optionLabelPath'
       context: Ember.computed.alias 'controller.selection'
+      ###*
+      * This is to override the click event in the SelectOptionView
+      * so that we can make sure only when we click on a new item in the list,
+      * the userDidSelect action will be fired, not when we open the dropdown.
+      * @override
+      ###
+      click: -> Ember.K()
```
Comment (embooglement) [inline src/select.coffee]: `click: Ember.K`

File: src/select.coffee [hunk 2]
```diff
@@ -209,6 +209,13 @@ Ember.AddeparMixins.ResizeHandlerMixin, Ember.Widgets.KeyboardHelper,
       tagName: 'span'
       labelPath: Ember.computed.alias 'controller.optionLabelPath'
       context: Ember.computed.alias 'controller.selection'
+      ###*
+      * This is to override the click event in the SelectOptionView
+      * so that we can make sure only when we click on a new item in the list,
+      * the userDidSelect action will be fired, not when we open the dropdown.
```
Comment (philpee2) [inline src/select.coffee]: I would add some clarification here that this view for the selected item is an extension of the view used to display each option in the popover list, and we don't want the click handler for the options in the popover list to be triggered when you click on the selected item. It's not really about the `userDidSelect` action. That just happens to be one of the effects of the click handler

**Comment by cyril-sf** [comment]
we need to figure out why the tests are failing.

**Comment by philpee2** [comment]
Hm, they were passing locally. Is that an old run?

**Comment by cyril-sf** [comment]
I don't think so

**Comment by embooglement** [comment]
They are failing locally when running the entire module, so this is probably a test pollution problem.

File: src/select.coffee [hunk 3]
```diff
@@ -209,6 +209,17 @@ Ember.AddeparMixins.ResizeHandlerMixin, Ember.Widgets.KeyboardHelper,
       tagName: 'span'
       labelPath: Ember.computed.alias 'controller.optionLabelPath'
       context: Ember.computed.alias 'controller.selection'
+      ###*
+      * Note: This view is an extension of the view used to display used to
```
Comment (philpee2) [inline src/select.coffee]: `used to display used to display`

File: src/select.coffee [hunk 4]
```diff
@@ -209,6 +209,17 @@ Ember.AddeparMixins.ResizeHandlerMixin, Ember.Widgets.KeyboardHelper,
...
+      * display each option in the dropdown list.
+      * We would like to override the click event in the SelectOptionView
+      * so that we can make sure that only when we click on a new option in the
+      * list, the click event in the SelectOptionView is fired, not when we open
+      * the dropdown. It will allow us to trigger the change item event when we
+      * click to select a new option in the dropdown.
+      * @override
+      ###
+      click: -> Ember.K
```
Comment (philpee2) [inline src/select.coffee]: Can this need to be `click: Ember.K`?

File: tests/unit/select_test.coffee [hunk 5]
```diff
@@ -122,6 +121,32 @@ test 'Test keyboard interaction', ->
       pressing ESC')
     validateFocus('Select component should be focused after pressing ESC')
 
+test 'Test userSelected action', ->
+  expect 3
+
+  select = @subject
```
Comment (philpee2) [inline tests/unit/select_test.coffee]: Does any of this need to be torn down to prevent test pollution?

File: src/select.coffee (hunk 4 above)
Comment (jordansmith42) [inline src/select.coffee]: It's a noop http://emberjs.com/api/#method_K

File: src/select.coffee (hunk 3 above)
Comment (thangdinh) [inline src/select.coffee]: Done

File: src/select.coffee (hunk 4 above)
Comment (thangdinh) [inline src/select.coffee]: I think we are always use `click: Ember.K`?

File: tests/unit/select_test.coffee (hunk 5 above)
Comment (jordansmith42) [inline tests/unit/select_test.coffee]: Looks like @cyril-sf already merged his fix for the test pollution

**Pull request merged**